src/unit.rs

```diff
@@ -824,6 +824,26 @@ impl<'abbrev, 'unit, R: Reader> DebuggingInformationEntry<'abbrev, 'unit, R> {
     pub fn attr_value(&self, name: constants::DwAt) -> Result<Option<AttributeValue<R>>> {
         self.attr(name).map(|attr| attr.map(|attr| attr.value()))
     }
+
+    /// Parse an entry. Returns `Ok(None)` for null entries.
+    fn parse(input: &mut R,
+             unit: &'unit UnitHeader<R>,
+             abbreviations: &'abbrev Abbreviations)
+             -> Result<Option<Self>> {
+        let offset = unit.header_size() + input.offset_from(&unit.entries_buf);
+        let code = input.read_uleb128()?;
+        if code == 0 {
+            return Ok(None);
```

Comment:
Slight aside: Does it make sense to delay calculating the offset until after reading the code? Maybe the compiler can reorder these operations anyways, and it isn't worth it.